### PR TITLE
Add Prelude/Optional/concatMap

### PR DESCRIPTION
**Pull request by stefanobaghino**
As suggested here: https://stackoverflow.com/a/64094498/3314107

### CODE REVIEW

**Review by Gabriel439 — APPROVED** [review]
Thank you!  I'll merge after the review period is complete (assuming no objections)

File: Prelude/Optional/concatMap [hunk 1]
```diff
@@ -0,0 +1,2 @@
+  ./concatMap.dhall sha256:c06ae93e900a99299c9b4ec4912ca5395a0103ea88292205b08dd7e345632818
+? ./concatMap.dhall
```
Comment (sjakobi) [inline Prelude/Optional/concatMap:2]: Is there a reason for adding a version without the `.dhall` extension? E.g. `Prelude.Map.unpackOptionals` was added without such a version in https://github.com/dhall-lang/dhall-lang/pull/1056.

File: Prelude/Optional/concatMap.dhall [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    : ∀(a : Type) → ∀(b : Type) → (a → Optional b) → Optional a → Optional b
+    = λ(a : Type) →
+      λ(b : Type) →
+      λ(f : a → Optional b) →
+      λ(o : Optional a) →
+        merge { Some = f, None = None b } o
+
+let exampleFun
+    : Natural → Optional Natural
+    = λ(n : Natural) → if Natural/even n then Some (n + 1) else None Natural
```
Comment (sjakobi) [inline Prelude/Optional/concatMap.dhall:12]: I believe `dhall-docs` doesn't currently include such additional let-bindings in the examples sections. Not sure what should be done about it though.

CC @german1608.

File: Prelude/Optional/concatMap (hunk 1 above)
Comment (stefanobaghino) [reply Prelude/Optional/concatMap:2]: I followed the pattern I saw for other functions. My knowledge of Dhall is very limited, but I assume this is a convenience for importing. How does one tell whether or not to add the version without the `.dhall` extension?

Comment (sjakobi) [reply Prelude/Optional/concatMap:2]: I think the files without the `.dhall` extension were simply kept for backward compatibility when we added the `.dhall` extensions in https://github.com/dhall-lang/dhall-lang/pull/1026. I'm not aware of a reason for adding such versions for new additions though.

Comment (stefanobaghino) [reply Prelude/Optional/concatMap]: I see, thanks, addressed by https://github.com/dhall-lang/dhall-lang/pull/1076/commits/4af394d5e1696a68ace5f6bee7670f5aa1d95d26.

**Review by sjakobi — APPROVED** [review]
Thanks!

**Pull request merged**